### ISSUE TITLE
[all versions] Dipatcher events are not reliable

### ISSUE THREAD

**Issue by SeriousBarbie** [OldUnreal/UnrealTournamentPatches#1454]
When Dispatcher has to process a a lot of events it swallows most of them. I guess because `OutEvents` are executed in a state code where `function Trigger()` is disabled.
Example to reproduce: enter attached map and kill the 50 pupaes with the WarheadLauncher what produces 50 events `PupaeDied' in a short time.
Expected: both counter (NumToCount=50) complete.
Actual: only a few events are raised for the Counters.
[SwallowedEvents.unr.zip](https://github.com/OldUnreal/UnrealTournamentPatches/files/13401419/SwallowedEvents.unr.zip)
![SwallowedEvents unr](https://github.com/OldUnreal/UnrealTournamentPatches/assets/81397053/e1d8f512-2b58-4f67-ac52-c1fe76d2d23a)

**Comment by SeriousBuggie** [comment]
This can be solved by not call Sleep if time is zero, however, this can break unknown count of mods and maps, so not acceptable.
If you need use dispatcher in such way - you need use own copy, where sleep call not called if delay time is zero.

**Comment by SeriousBarbie** [comment]
My idea was to not disabling the trigger function but creating a queue that stores incoming events for executing them after each other.

**Comment by SeriousBuggie** [comment]
Yes. It can be documented. About queue - some maps in loop call dispathcer of self. If not ignore that, it be endless loop.
Which goes to crash, additional load or waste cpu/memory.
Queue there dangerous. Same as ignore Sleep call.

**Comment by an-eternity** [comment]
There are maps that in some cases can't be finished because of this issue... (Probably, mappers didn't consider such cases may occur)

It is possible to automatically detect which way `Dispatcher` is being used in... If exactly such case where a modified `Dispatcher` is needed detected (such as this one - where 2 or more Scripted Pawns as an input trigger event source, and a Counter as an output event target), then standard `Dispatcher` gets automatically replaced by a modified one which is able to handle multiple events at one.
Otherwise, need to manually check every map and apply/enable a modified `Dispatcher` replacement fix if needed.

**Comment by SeriousBuggie** [comment]
It must be manually check and replace every map.
Because dispatcher can make self-loop not directly, but with mediators. Which mean such impossible to track.
Global changes must be safe.
